llvm/lib/SYCLLowerIR/SYCLPropagateAspectsUsage.cpp

```diff
@@ -58,22 +59,19 @@ TypeToAspectsMapTy getTypesThatUseAspectsFromMetadata(const Module &M) {
     return Result;
 
   LLVMContext &C = M.getContext();
-  for (const auto OperandIt : Node->operands()) {
-    const MDNode &N = *OperandIt;
-    assert(N.getNumOperands() > 1 && "intel_types_that_use_aspect metadata "
-                                     "shouldn't contain empty metadata nodes");
+  for (const MDNode *N : Node->operands()) {
+    assert(N->getNumOperands() > 1 && "intel_types_that_use_aspect metadata "
+                                      "shouldn't contain empty metadata nodes");
 
-    const auto *TypeName = cast<MDString>(N.getOperand(0));
+    const auto *TypeName = cast<MDString>(N->getOperand(0));
     const Type *T = StructType::getTypeByName(C, TypeName->getString());
     assert(T &&
            "invalid type referenced by intel_types_that_use_aspect metadata");
 
     AspectsSetTy &Aspects = Result[T];
-    for (size_t I = 1; I != N.getNumOperands(); ++I) {
-      const auto *CAM = cast<ConstantAsMetadata>(N.getOperand(I));
-      const Constant *C = CAM->getValue();
-      Aspects.insert(cast<ConstantInt>(C)->getSExtValue());
-    }
+    for (const MDOperand &Op : drop_begin(N->operands()))
+      Aspects.insert(cast<ConstantInt>(cast<ConstantAsMetadata>(Op)->getValue())
+                         ->getSExtValue());
```

Comment:
Personally, I find this chain a bit too long, but I won't object. As an alternative/compromise:
```suggestion
      const Constant *C = cast<ConstantAsMetadata>(Op)->getValue()
      Aspects.insert(cast<ConstantInt>(C)->getSExtValue());
```